// vue-recaptcha/dist/api.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ChallengeV2", {
  enumerable: true,
  get: function () {
    return _ChallengeV.default;
  }
});
Object.defineProperty(exports, "ChallengeV3", {
  enumerable: true,
  get: function () {
    return _ChallengeV2.default;
  }
});
Object.defineProperty(exports, "Checkbox", {
  enumerable: true,
  get: function () {
    return _Checkbox.default;
  }
});
Object.defineProperty(exports, "RecaptchaV2State", {
  enumerable: true,
  get: function () {
    return _challengeV.RecaptchaV2State;
  }
});
Object.defineProperty(exports, "createPlugin", {
  enumerable: true,
  get: function () {
    return _plugin.createPlugin;
  }
});
Object.defineProperty(exports, "defineScriptLoader", {
  enumerable: true,
  get: function () {
    return _common.defineScriptLoader;
  }
});
Object.defineProperty(exports, "toQueryString", {
  enumerable: true,
  get: function () {
    return _common.toQueryString;
  }
});
Object.defineProperty(exports, "useChallengeV2", {
  enumerable: true,
  get: function () {
    return _challengeV.useChallengeV2;
  }
});
Object.defineProperty(exports, "useChallengeV3", {
  enumerable: true,
  get: function () {
    return _challengeV2.useChallengeV3;
  }
});
Object.defineProperty(exports, "useRecaptchaContext", {
  enumerable: true,
  get: function () {
    return _context.useRecaptchaContext;
  }
});
Object.defineProperty(exports, "useRecaptchaProvider", {
  enumerable: true,
  get: function () {
    return _scriptProvider.useRecaptchaProvider;
  }
});
Object.defineProperty(exports, "useRecaptchaProxy", {
  enumerable: true,
  get: function () {
    return _context.useRecaptchaProxy;
  }
});
var _context = require("./composables/context.cjs");
var _challengeV = require("./composables/challenge-v2.cjs");
var _challengeV2 = require("./composables/challenge-v3.cjs");
var _scriptProvider = require("./composables/script-provider.cjs");
var _ChallengeV = _interopRequireDefault(require("./components/ChallengeV2.vue"));
var _ChallengeV2 = _interopRequireDefault(require("./components/ChallengeV3.vue"));
var _Checkbox = _interopRequireDefault(require("./components/Checkbox.vue"));
var _plugin = require("./plugin.cjs");
var _common = require("./script-manager/common.cjs");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
